homeassistant/components/calendar/__init__.py

```diff
@@ -137,7 +137,7 @@ def is_offset_reached(
     return start + offset_time <= dt.now(start.tzinfo)
 
 
-class CalendarEventDevice(Entity):
+class CalendarEntity(Entity):
```

Comment:
Should we add a compatibility class with the old name which logs a warning if used as we've done before when changing entity class names?

Reply:
Can we keep the old class, without changes, in parallel with the new class and add a compatibility layer in the integration so that both can be used? Log a warning when using the old class.

We've already done the work for the core integrations, so I don't think we need to undo that.

Reply:
But you're of course free to split the work as you see best.

Reply:
Got it, leaving the old class is much simpler, though i think we have to also keep these helper functions. The compatibility shim really just seems like it need to be for the API, which is simple.  I'm punting on updating behavior for the old APIs. So we'll keep this crufty code around longer for a few more releases, but definitely safer rollout. I added a demo calendar to exercise this since I don't really trust anything unless its tested.